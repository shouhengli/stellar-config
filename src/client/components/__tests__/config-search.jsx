const React = require('react');
const renderer = require('react-test-renderer');
const { shallow } = require('enzyme');
const Search = require('../config-search.jsx');

describe('component config-search', () => {
  let props;

  beforeEach(() => {
    props = {
      Tab: 'div',
      ActiveTab: 'div',
      Item: 'div',
      searchText: 'default',
      configTypes: ['source', 'mapping', 'graphSchema'],
      activeConfigType: 'mapping',
      configNames: ['default', 'finance', 'hr'],
      handleSearchTextChange: jest.fn(),
      handleHideButtonClick: jest.fn()
    };
  });

  test('can be rendered', () => {
    const tree = renderer.create(<Search {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('has changeable search field', () => {
    const wrapper = shallow(<Search {...props} />);
    const event = { target: { value: 'def' } };
    wrapper.find('input').simulate('change', event);

    expect(props.handleSearchTextChange).toHaveBeenCalledTimes(1);
    expect(props.handleSearchTextChange).toHaveBeenCalledWith(event);
  });

  test('can be hidden', () => {
    const wrapper = shallow(<Search {...props} />);
    wrapper.find('button').simulate('click');

    expect(props.handleHideButtonClick).toHaveBeenCalledTimes(1);
  });
});
