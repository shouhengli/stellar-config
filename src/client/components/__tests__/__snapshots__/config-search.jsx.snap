// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component config-search can be rendered 1`] = `
<nav
  className="panel config-search"
>
  <div
    className="panel-block config-search-input"
  >
    <div
      className="field has-addons"
    >
      <div
        className="control has-icons-left is-expanded"
      >
        <input
          className="input is-medium"
          onChange={[Function]}
          type="text"
          value="default"
        />
        <span
          className="icon is-left"
        >
          <i
            className="fa fa-search"
          />
        </span>
      </div>
      <div
        className="control"
      >
        <button
          className="button is-medium"
          onClick={[Function]}
        >
          <span
            className="icon is-small"
          >
            <i
              className="fa fa-times"
            />
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="panel-block"
    onClick={[Function]}
  >
    <span
      className="panel-icon"
    >
      <i
        className="fa fa-book"
      />
    </span>
    default
  </div>
</nav>
`;
